### PR TITLE
upgrade accountin to 17 (not compeleted but is a good starting point)

### PR DESCRIPTION
**Pull request by hamidk1010**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Thanks for the contribution.

Please preserve commit history following technical method explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0.

If the jump is between several versions, you have to modify the source branch in the main command to accommodate it to this circumstance.

File: l10n_ir_account/data/template/account.account-ir.csv [hunk 1]
```diff
@@ -384,4 +384,3 @@
 "account_account_template_0_911108","سایر اسناد تضمینی به عهده شرکت","911108","off_balance","","0"
 "account_account_template_0_911201","طرف حسابهای انتظامی به عهده شرکت","911201","off_balance","","0"
 "account_account_template_0_911202","طرف حسابهای انتظامی به نفع شرکت","911202","off_balance","","0"
```
Comment (saeed-raeisi) [inline l10n_ir_account/data/template/account.account-ir.csv]: Hi
here name must be in English and then translated in the po, and the id could be shorter
https://github.com/OCA/l10n-iran/blob/15.0/l10n_ir/data/account.account.template.csv

File: l10n_ir_accounting/data/res.bank.csv [hunk 2]
```diff
@@ -0,0 +1,24 @@
+id,name,bic,country/id,zip,city,street,email,phone
```
Comment (saeed-raeisi) [inline l10n_ir_accounting/data/res.bank.csv]: same here
https://github.com/OCA/l10n-iran/blob/15.0/l10n_ir/data/res.bank.csv

File: l10n_ir_account/data/template/account.account-ir.csv (hunk 1 above)
Comment (hamidk1010) [reply l10n_ir_account/data/template/account.account-ir.csv]: hi,this is not mandatory.
this iran chart also must be refine in future,we can do it later
https://github.com/odoo/odoo/blob/17.0/addons/l10n_ua/data/template/account.account-ua_ias.csv

now tax,tax group , fiscal position and default accounts have priority to do  first

- [x] tax
- [x] tax group
- [x] fiscal position
- [ ] default accounts

@saeed-raeisi

Comment (hamidk1010) [reply l10n_ir_account/data/template/account.account-ir.csv]: also translation must be done with adding a column to same file with this name: column to be translate@language code
like name@fa  or description@fa
https://github.com/odoo/odoo/blob/17.0/addons/l10n_sk/data/template/account.account-sk.csv

**Comment by dreispt** [comment]
Thank you @hamidk1010 
The migration PR needs to keep the module's commit history.
You can find a detailed how to here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#technical-method-to-migrate-a-module-from-160-to-170-branch
Follow it carefully.

Also, we don't want multiple fix or migration steps.
You should either squash later fixes into a single commit, or just amend the last commit as you go, using `git commit --amend --no-edit`.

**Comment by hamidk1010** [comment]
i did it: https://github.com/OCA/l10n-iran/pull/35 @dreispt

**Comment by hamidk1010** [comment]
continue here: https://github.com/OCA/l10n-iran/pull/35

**Pull request closed without merging**